libtuf-server/src/main/scala/com/advancedtelematic/libtuf_server/data/Entities.scala

```diff
@@ -0,0 +1,16 @@
+package com.advancedtelematic.libtuf_server.data
+
+import com.advancedtelematic.libtuf.data.TufDataType
+import com.advancedtelematic.libtuf.data.TufCodecs._
+import io.circe.{Decoder, Encoder}
+import io.circe.generic.semiauto._
+
+object Entities {
+
+  object KeyType {
+    implicit val encoder: Encoder[KeyType] = io.circe.generic.semiauto.deriveEncoder
+    implicit val decoder: Decoder[KeyType] = io.circe.generic.semiauto.deriveDecoder
+  }
+
+  case class KeyType(keyType: TufDataType.KeyType)
```

Comment:
Should be named something like `CreateRepositoryRequest(keyType: KeyType)`, I think we use that pattern somewhere else.

Reply:
Hmm, that would make it sound to me like there being a generic endpoint to which we would send specific requests. How about `CreateRepositoryParam` or `KeyTypeParam`?

Reply:
What do you mean? We do have an endpoint where you need to send specific requests, POST on /root

Reply:
yes, so?

Reply:
I mean that this new thing isn't a request, just an optional part of it, that's why I don't want to call it "Request".

Reply:
How is it not a request? It's not an http request but from the user it's a request. They want to create something and build an object requesting it.

This is used everywhere in the code base already. 

- ClientRootGenRequest
- RequestTargetItem
- MultiTargetUpdateRequest
- CreateDiffInfoRequest
- FileCacheRequest
- TargetUpdateRequest
- PaginationRequest

And probably others.

Reply:
No they don't need an object to request it, the new parameter is optional. But I can call it `CreateRepositoryRequest`, no problem.